### PR TITLE
Integrate Stripe payment processing with API

### PR DESCRIPTION
**Pull request by wmemorgan**
https://trello.com/c/DgnTT2ur

### CODE REVIEW

**Comment by wmemorgan** [comment]
[Integrate Stripe payment processing with API](https://trello.com/c/DgnTT2ur/140-integrate-stripe-payment-processing-with-api)

File: data/migrations/20191025101146_tickets_add_schedule_id.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+exports.up = async function(knex) {
```
Comment (GannonDetroit) [inline data/migrations/20191025101146_tickets_add_schedule_id.js]: migrations look good

File: routes/salesRoutes.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
 const router = require("express").Router();
-
+const stripe = require("stripe")(process.env.STRIPE_SECRET_KEY);
```
Comment (GannonDetroit) [inline routes/salesRoutes.js]: thank you for using env variable for this lol

File: routes/salesRoutes.js [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
 const db = require("../data/models");
 
+const stripeChargeCallback = res => async (stripeErr, stripeRes) => {
```
Comment (GannonDetroit) [inline routes/salesRoutes.js]: looks good

File: routes/salesRoutes.js [hunk 4]
```diff
@@ -76,31 +86,59 @@ router.put("/:saleId", async (req, res, next) => {
...
-
-	if (!ticket_id || !sale_amount) {
-		res.status(400).json({ message: "All fields are required." });
-	}
-
-	if (typeof sale_amount === "string") {
-		user_id = parseFloat(user_id, 10);
-	}
-
+router.post("/", async (req, res) => {
```
Comment (GannonDetroit) [inline routes/salesRoutes.js]: looks good for what our needs are.

File: routes/salesRoutes.js [hunk 5]
```diff
@@ -76,31 +86,59 @@ router.put("/:saleId", async (req, res, next) => {
...
+		if (charge) {
+			console.log("Success", charge);
+			res.send({ success: charge });
+		} else throw error;
+	} catch (error) {
+		console.log("ERR MSG: ", error);
+		res.status(500).send({ message: "Internal server error.", error: error });
 	}
+
+	// const body = {
```
Comment (GannonDetroit) [inline routes/salesRoutes.js]: is there a reason we commented this out instead of just deleting it? If not then lets delete it to keep the code clean

Comment (wmemorgan) [reply routes/salesRoutes.js]: That was the previous code before the Stripe integration. I'll clean that up before merging the PR.

**Pull request merged**